src/backend/cuda/kernel/topk.hpp

```diff
@@ -56,15 +57,16 @@ static __global__ void kerTopkDim0(Param<T> ovals, Param<uint> oidxs,
     uint* ires = oidxs.ptr + by * oidxs.strides[1] + bz * oidxs.strides[2] +
                  bw * oidxs.strides[3];
 
-    T keys[TOPK_IDX_THRD_LOAD];
+    compute_t<T> keys[TOPK_IDX_THRD_LOAD];
     ValueType vals[TOPK_IDX_THRD_LOAD];
 
     for (uint li = 0, i = gx; li < TOPK_IDX_THRD_LOAD; i += gxStride, li++) {
         if (i < elements) {
             keys[li] = kdata[i];
```

Comment:
keys[li] = (compute_t<T>)kdata[i];
is a must have to compile under old gcc and/or cuda10.1